### ISSUE TITLE
Question: How to extract the productive head sequence only from a clonotype

### ISSUE THREAD

**Issue by slowsmile** [milaboratory/mixcr#684]
Dear all

I've studied MIXCR tool for a while but I am struggling to find a solution for my current task. Basically, I want to extract a representative sequence from the BCR clones (one sequence for each clone). This may be equivalent to extracting full sequence of the "head node" of each clonetype. I tried several ways:

1) "**mixcr exportClones**" only output targetSequences for each clone, in my case, mixcr assembled only to CDR3 region so the target sequence only has CDR3 region. However, I'd like to extract the full read sequence from that clone

2) "**mixcr exportReadsForClones**" can export the original fastq reads for each clone but it outputs all reads sequences, and in my case I only want the read sequence of the head node.

I understand that mixcr assemble clones in a hierachical tree structure, designating the parental clonotype as the clones of the highest counts. So is there a way to extract the "**head clonotype full sequence**".  Also, is there a format to show which sequence is parental or child in a clonotype?

Thanks a lot
Xiao

**Comment by PoslavskySV** [comment]
Dear Xiao,

did you try contig assembly? https://mixcr.readthedocs.io/en/develop/assembleContigs.html

Best,
Stanislav

**Comment by slowsmile** [comment]
Thanks a lot Stanislav for your quick response!I ran contig assembly and
exported full BCR receptor using "mixcr exportClones -c IG -p fullImputed
full_clones.clns full_clones.txt". I assume the targetSequences column of
the corresponding clone should contain the representative sequence of that
clone. However, One follow-up question is that I sometimes observed three
Targetsequences separated by comma, which baffled me a lot.

In my case, I have paired-end reads (may not overlap) so I assume I should
have two targetsequences for each clone, isn't it? Each
targetsequence corresponds to the aligned sequence of one read. But why do
we have more than 2 Target sequences with full BCR contig assembly?

Regards

Xiao
